### PR TITLE
Follow sampling rate setting

### PR DESCRIPTION
**Pull request by Oscil8**
Noticed that the sample rate setting has no effect on what gets sent to the collector; need to use ::Trace.should_sample? from the finagle-thrift gem.

### CODE REVIEW

File: zipkin-gems/zipkin-tracer/lib/zipkin-tracer.rb [hunk 1]
```diff
@@ -55,9 +55,9 @@ def initialize(app)
     end
 
     def call(env)
+      ::Trace.sample_rate=(@sample_rate)
       id = get_or_create_trace_id(env)
       ::Trace.default_endpoint = ::Trace.default_endpoint.with_service_name(@service_name).with_port(@service_port)
-      ::Trace.sample_rate=(@sample_rate)
```
Comment (franklinhu) [inline zipkin-gems/zipkin-tracer/lib/zipkin-tracer.rb:60]: We may as well move all the variable setting to the beginning of the method for consistency. Also, let's make a note that get_or_create_trace_id depends on sample_rate being already set.

Comment (Oscil8) [inline zipkin-gems/zipkin-tracer/lib/zipkin-tracer.rb:60]: Sounds good, I'll update my patch

**Comment by Oscil8** [comment]
@franklinhu Let me know if you've got any additional comments.

**Comment by franklinhu** [comment]
+1

**Comment by sprsquish** [comment]
Pulled internally. Thanks!

**Pull request closed without merging**